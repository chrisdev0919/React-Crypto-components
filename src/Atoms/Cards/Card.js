import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const BaseCard = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: ${props => props.height};
  width: ${props => props.width};
  box-shadow: 0px 0px 25px 5px ${props => props.theme['gray-1']};
`

const Card = ({ children, height, width }) => (
  <BaseCard height={height} width={width}>
    {children}
  </BaseCard>
)

Card.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string
}

Card.defaultProps = {
  height: 'auto',
  width: 'auto'
}

export default Card
