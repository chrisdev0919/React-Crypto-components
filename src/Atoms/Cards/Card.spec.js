import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Card from './Card'

describe('Card component', () => {
  it('renders correctly', () => {
    const component = shallow(<Card>MyCard</Card>)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
