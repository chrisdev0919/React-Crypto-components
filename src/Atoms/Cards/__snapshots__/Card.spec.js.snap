// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<styled.div
  height="auto"
  width="100%"
>
  MyCard
</styled.div>
`;
